lib/pavilion/schedulers/plugins/flux.py

```diff
@@ -0,0 +1,291 @@
+# pylint: disable=too-many-lines
+"""The Flux Framework Scheduler Plugin."""
+
+import os
+import time
+from typing import List, Union, Any, Tuple
+
+import yaml_config as yc
+from pavilion import sys_vars
+from pavilion.jobs import Job, JobInfo
+from pavilion.status_file import STATES, TestStatusInfo
+from pavilion.types import NodeInfo, NodeList
+from ..advanced import SchedulerPluginAdvanced
+from ..config import validate_list
+
+
+# Just import flux once
+try:
+    import flux
+    import flux.job
+    import flux.resource
+    from flux.job import JobspecV1
+except ImportError:
+    flux = None
+
+flux_states = [
+    "DEPEND",
+    "SCHED",
+    "RUN",
+    "CLEANUP",
+    "COMPLETED",
+    "FAILED",
+    "CANCELED",
+    "TIMEOUT",
+]
```

Comment:
The equivalent in the Slurm scheduler was about figuring out the state of each node, but these look like job states. If that's what you're using them for, that's fine.